internal/comparetest/metrics_test.go

```diff
@@ -428,14 +428,16 @@ func TestCompareMetrics(t *testing.T) {
 				IgnoreResourceAttributeValue("node_id"),
 			},
 			withoutOptions: expectation{
-				err: multierr.Combine(
-					errors.New("missing expected resource with attributes: map[namespace:BB903-test node_id:BB903-expected]"),
-					errors.New("missing expected resource with attributes: map[namespace:BB904-test node_id:BB904-expected]"),
-					errors.New("missing expected resource with attributes: map[namespace:BB902-test node_id:BB902-expected]"),
-					errors.New("extra resource with attributes: map[namespace:BB903-test node_id:BB903-actual]"),
-					errors.New("extra resource with attributes: map[namespace:BB902-test node_id:BB902-actual]"),
-					errors.New("extra resource with attributes: map[namespace:BB904-test node_id:BB904-actual]"),
-				),
+				validator: func(t *testing.T, err error) {
+					errs := multierr.Errors(err)
+					require.Equal(t, 6, len(errs))
+					require.Contains(t, errs, errors.New("missing expected resource with attributes: map[namespace:BB903-test node_id:BB903-expected]"))
+					require.Contains(t, errs, errors.New("missing expected resource with attributes: map[namespace:BB904-test node_id:BB904-expected]"))
+					require.Contains(t, errs, errors.New("missing expected resource with attributes: map[namespace:BB902-test node_id:BB902-expected]"))
+					require.Contains(t, errs, errors.New("extra resource with attributes: map[namespace:BB903-test node_id:BB903-actual]"))
+					require.Contains(t, errs, errors.New("extra resource with attributes: map[namespace:BB902-test node_id:BB902-actual]"))
+					require.Contains(t, errs, errors.New("extra resource with attributes: map[namespace:BB904-test node_id:BB904-actual]"))
```

Comment:
The order is declared by how the ResourceMetrics are sorted. It reshuffles every time the hash function changes. I can change the order this time, but it will need to be done again if we change the hash function later.

Reply:
Seems like the hash function will be stable soon enough, if not with this change, so should we just update the order this time instead of adding the validator mechanism?

Reply:
Removed the validator